### PR TITLE
[operator/nodejs] document disabling node auto-instrumentations

### PR DESCRIPTION
**Pull request by JamieDanielson**
### Summary

OTel JS [auto-instrumentations-node v0.48.0](https://github.com/open-telemetry/opentelemetry-js-contrib/releases/tag/auto-instrumentations-node-v0.48.0) now includes the ability to disable instrumentations using environment variables.

### Changes

- Add section to operator docs for disabling instrumentations
- Add headers and section to zero-code js configuration page for disabling instrumentations

I'm not sure if this is too redundant in this text, as it's mostly a copy-paste. Open to suggestions!

### CODE REVIEW

File: content/en/docs/kubernetes/operator/automatic.md
Comment (TylerHelmuth) [inline content/en/docs/kubernetes/operator/automatic.md:317]: Lets add a second package to show the comma-separated list in action.

Comment (JamieDanielson) [reply content/en/docs/kubernetes/operator/automatic.md]: Added grpc for new example `fs,grpc`

**Review by TylerHelmuth — APPROVED** [review]

**Review by trentm — APPROVED** [review]

**Comment by JamieDanielson** [comment]
Wait don't merge this I have a typo on a link

**Comment by JamieDanielson** [comment]
> Wait don't merge this I have a typo on a link

Fixed!

Side note: I noticed in a separate issue that there is external link checking but it is currently excluding GitHub URLs, which is how this one got missed. Noting for anyone curious (I was 😄 )

File: content/en/docs/kubernetes/operator/automatic.md
Comment (chalin) [inline content/en/docs/kubernetes/operator/automatic.md:306]:
```suggestion
variable. For details, see 
[Excluding instrumentation libraries](/docs/zero-code/js/configuration/#excluding-instrumentation-libraries).
```

**Review by chalin — COMMENTED** [review]
LGTM. See inline suggestion.

Also, what happens if both `OTEL_NODE_ENABLED_INSTRUMENTATIONS` and `OTEL_NODE_DISABLED_INSTRUMENTATIONS` are provided? Should there be a note explaining this?

**Comment by chalin** [comment]
> Thanks for fixing those links! I noticed the broken http link as well. (@chalin not sure if this needs attention or if we're happy to continue ignoring GH links.)

Yes, thanks for fixing the link. I don't have the time to enable the link checking for GH links atm. If it becomes problematic (i.e., we feel that we've been manually detecting too many broken links into GH repos), then we can bump the priority on enabling such link checks.

**Review by david-luna — APPROVED** [review]

**Comment by JamieDanielson** [comment]
> Also, what happens if both `OTEL_NODE_ENABLED_INSTRUMENTATIONS` and `OTEL_NODE_DISABLED_INSTRUMENTATIONS` are provided? Should there be a note explaining this?

All instrumentations are disabled because they are mutually exclusive. I can add that note. I've opened a [PR on js-contrib](https://github.com/open-telemetry/opentelemetry-js-contrib/pull/2325) to illustrate this behavior and if there is feedback to change behavior or update the debug message I can come back and open a new PR for that.

File: content/en/docs/kubernetes/operator/automatic.md
Comment (tiffany76) [inline content/en/docs/kubernetes/operator/automatic.md:322]:
```suggestion
> all instrumentations are disabled.
```

File: content/en/docs/zero-code/js/configuration.md
Comment (tiffany76) [inline content/en/docs/zero-code/js/configuration.md:91]:
```suggestion
> all instrumentations are disabled.
```

**Review by tiffany76 — APPROVED** [review]
We try to avoid future tense if possible, so there's no question of _when_ the thing happens. Otherwise, LGTM! Thanks.

File: content/en/docs/kubernetes/operator/automatic.md
Comment (JamieDanielson) [reply content/en/docs/kubernetes/operator/automatic.md]: Ah I pushed a commit changing the note type so it didn't let me commit this suggestion directly 😅  Made this change though, thanks for the explanation!

**Comment by JamieDanielson** [comment]
> > Also, what happens if both `OTEL_NODE_ENABLED_INSTRUMENTATIONS` and `OTEL_NODE_DISABLED_INSTRUMENTATIONS` are provided? Should there be a note explaining this?
> 
> All instrumentations are disabled because they are mutually exclusive. I can add that note. I've opened a [PR on js-contrib](https://github.com/open-telemetry/opentelemetry-js-contrib/pull/2325) to illustrate this behavior and if there is feedback to change behavior or update the debug message I can come back and open a new PR for that.

@chalin I added the note for behavior of both env vars being set, thanks for calling it out!

File: content/en/docs/kubernetes/operator/automatic.md
Comment (trentm) [inline content/en/docs/kubernetes/operator/automatic.md:324]: Are you sure this is correct? My understanding is that if both are set, then first `*_ENABLED_*` is used to establish a starting set, then any of those listed in `*_DISABLED_*` are removed.

```
% cd ~/src/opentelemetry-js-contrib/metapackages/auto-instrumentations-node
% node
> var mod = require('./')
undefined
> process.env.OTEL_NODE_ENABLED_INSTRUMENTATIONS = 'dns,fs,hapi'
'dns,fs,hapi'
> mod.getNodeAutoInstrumentations().map(instr => instr.instrumentationName)
[
  '@opentelemetry/instrumentation-dns',
  '@opentelemetry/instrumentation-fs',
  '@opentelemetry/instrumentation-hapi'
]
> process.env.OTEL_NODE_DISABLED_INSTRUMENTATIONS = 'bunyan,fs'
'bunyan,fs'
> mod.getNodeAutoInstrumentations().map(instr => instr.instrumentationName)
[
  '@opentelemetry/instrumentation-dns',
  '@opentelemetry/instrumentation-hapi'
]
```

Perhaps:

```suggestion
If both environment variables are specified, `OTEL_NODE_ENABLED_INSTRUMENTATIONS` is first used to establish a set of instrumentation package names, then any names from `OTEL_NODE_DISABLED_INSTRUMENTATIONS` are removed from that set. Any remaining instrumentation names are enabled.
```

File: content/en/docs/zero-code/js/configuration.md
Comment (trentm) [inline content/en/docs/zero-code/js/configuration.md:92]: Similar Q to above about the behaviour if both are specified.

**Review by pavolloffay — APPROVED** [review]

**Review by svrnm — CHANGES_REQUESTED** [review]
Assuming that @trentm's comments are blocking I put this in to not accidentally merge it

File: content/en/docs/kubernetes/operator/automatic.md
Comment (JamieDanielson) [reply content/en/docs/kubernetes/operator/automatic.md:324]: Yeah no you're totally correct, I'm not sure what I was doing when I wrote the original test. Will update there and then here

File: content/en/docs/zero-code/js/configuration.md
Comment (JamieDanielson) [reply content/en/docs/zero-code/js/configuration.md]: Updated!

File: content/en/docs/kubernetes/operator/automatic.md
Comment (JamieDanielson) [reply content/en/docs/kubernetes/operator/automatic.md]: Updated!

**Review by chalin — APPROVED** [review]
LGTM, and I'm glad for this to get merged as is for now, but:

- We might prefer the use of an `{{% alert %}}` short code instead of blockquote markdown
- @svrnm - do we want to pull out that note into a shortcode given that it's used verbatim in two place. This would avoid drift.

**Comment by chalin** [comment]
Will let @svrnm have the final call here before this get's merged.

**Review by trentm — APPROVED** [review]
It does seem like there is some precedent in the content to use `> ...` blockquote markdown syntax for side-notes / call-outs.

File: content/en/docs/kubernetes/operator/automatic.md
Comment (svrnm) [inline content/en/docs/kubernetes/operator/automatic.md:323]:
```suggestion
{{% alert title="Note" color="info" %}}

If both environment variables are set, `OTEL_NODE_ENABLED_INSTRUMENTATIONS` is
applied first, and then `OTEL_NODE_DISABLED_INSTRUMENTATIONS` is applied to
that list. Therefore, if the same instrumentation is included in both lists,
that instrumentation will be disabled.

{{% /alert %}}
```

File: content/en/docs/zero-code/js/configuration.md
Comment (svrnm) [inline content/en/docs/zero-code/js/configuration.md:92]:
```suggestion
{{% alert title="Note" color="info" %}}

If both environment variables are set, `OTEL_NODE_ENABLED_INSTRUMENTATIONS` is
applied first, and then `OTEL_NODE_DISABLED_INSTRUMENTATIONS` is applied to
that list. Therefore, if the same instrumentation is included in both lists,
that instrumentation will be disabled.

{{% /alert %}}
```

**Comment by svrnm** [comment]
> It does seem like there is some precedent in the content to use `> ...` blockquote markdown syntax for side-notes / call-outs.

Yes, but we try to replace them whenever we handle an existing file, eventually we want to use the shortcode everywhere since it gives us more flexibility in styling & structuring those call outs.

I will fix the notice and merge this PR

**Review by svrnm — APPROVED** [review]

**Comment by JamieDanielson** [comment]
> Yes, but we try to replace them whenever we handle an existing file, eventually we want to use the shortcode everywhere since it gives us more flexibility in styling & structuring those call outs.

Ah, good to know. I had that [and removed it ](https://github.com/open-telemetry/opentelemetry.io/pull/4824/commits/0e4df3a8957bf4751522ba9164d74a34af91e36a) since it seemed like it was no longer a note that needed major attention. I thought it might be a "downgraded" note 😅

**Pull request merged**
